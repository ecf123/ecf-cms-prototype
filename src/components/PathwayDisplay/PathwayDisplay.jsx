import "./PathwayDisplay.scss";
import hamburger from "../../assets/hamburger-icon.svg";
import grid from "../../assets/grid-icon.svg";
import Button from "../Button/Button";
import { ReactComponent as Pencil } from "../../assets/pen-icon.svg";
import { Link } from "react-router-dom";
import { useState } from "react";

const PathwayDisplay = () => {
  const [border, setBorder] = useState(true);

  const toggleDisplay = (display) => {
    setBorder(display);
  };

  return (
    <div className="pathway-display">
      <div className="pathway-display__container">
        <Link to="*">
          <img
            className="pathway-display__option"
            id={border && "grid-icon"}
            src={grid}
            alt="grid display option button"
            onClick={() => toggleDisplay(true)}
          />
        </Link>
        <Link to="*">
          <img
            className="pathway-display__option"
            id={!border && "list-icon"}
            src={hamburger}
            alt="list view option button"
            onClick={() => toggleDisplay(false)}
          />
        </Link>
      </div>
      <Link to="*" className="pathway-display__button">
        <Button
          style={"button black round-border medium"}
          imgStyle={"medium-img white-svg"}
          textStyle={"text medium-text black"}
          buttonText={"Add Course"}
        >
          <Pencil />
        </Button>
      </Link>
    </div>
  );
};

export default PathwayDisplay;
