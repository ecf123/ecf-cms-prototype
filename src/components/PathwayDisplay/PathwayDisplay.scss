@use "../../styles/_variables" as *;

.pathway-display {
    height: 76px;
    width: 1334px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 30px;
    background-color: #fcfcfd;

    &__container {
        display: flex;
        align-items: center;

        .pathway-display__option {
            margin: 10px;
        }
    }

    #grid-icon {
        padding: 10px;
        border: 1.5px solid $color-grey-light;
        border-radius: 11px;
        background-color: #ffffff;
    }

    &__button {
        text-decoration: none;
    }
}