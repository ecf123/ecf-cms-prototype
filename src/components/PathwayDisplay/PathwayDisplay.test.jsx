import { render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import { BrowserRouter } from "react-router-dom";
import PathwayDisplay from "./PathwayDisplay.jsx";

it("should render the grid icon", () => {
    render(<PathwayDisplay />, {wrapper: BrowserRouter});
    const image = screen.getByAltText("grid display option button");
    expect(image).toBeInTheDocument();
  });

  it("should render the list icon", () => {
    render(<PathwayDisplay />, {wrapper: BrowserRouter});
    const image = screen.getByAltText("list view option button");
    expect(image).toBeInTheDocument();
  });

  it("Should render the add course button", () => {
    render(<PathwayDisplay />, {wrapper: BrowserRouter});
    const button = screen.getByText("Add Course");
    expect(button).toBeInTheDocument();
  })

  it("Should render Pathway Display", () => {
    const {container} = customRender(
      <PathwayDisplay/>, {wrapper: BrowserRouter}
    )
    expect(container).toMatchSnapshot();

  })


