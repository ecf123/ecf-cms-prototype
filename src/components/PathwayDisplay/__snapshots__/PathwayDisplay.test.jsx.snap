// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Pathway Display 1`] = `
<div>
  <div
    class="pathway-display"
  >
    <div
      class="pathway-display__container"
    >
      <img
        alt="grid display option button"
        class="pathway-display__option"
        id="grid-icon"
        src="grid-icon.svg"
      />
      <img
        alt="list view option button"
        class="pathway-display__option"
        src="hamburger-icon.svg"
      />
    </div>
    <a
      class="pathway-display__button"
      href="/*"
    >
      <button
        class="button black round-border medium"
      >
        <div
          class="medium-img white-svg"
        >
          <svg>
            pen-icon.svg
          </svg>
        </div>
        <p
          class="text medium-text black"
        >
          Add Course
        </p>
      </button>
    </a>
  </div>
</div>
`;
